pkg/rancher-components/src/components/StringList/StringList.vue

```diff
@@ -0,0 +1,572 @@
+    },
+
+    /**
+     * Create a new item and insert in the items list
+     */
+    saveItem() {
+      if (this.value) {
+        const items = [
+          ...this.items,
+          this.value.trim(),
+        ] as string[];
+
+        if (findStringDuplicates(items, this.caseSensitive)) {
+          this.toggleError('duplicate', true, INPUT.create);
+
+          return;
+        }
+
+        this.updateItems(items);
+      }
+      this.toggleCreateMode(false);
+    },
+
+    /**
+     * Update an existing item in the items list
+     */
+    updateItem(item: string) {
+      if (this.value) {
+        const items = [...this.items] as string[];
+        const index = findStringIndex(items, item, false);
+
+        if (index !== -1) {
+          items[index] = this.value.trim();
+        }
+
+        if (findStringDuplicates(items, this.caseSensitive)) {
+          this.toggleError('duplicate', true, INPUT.edit);
+
+          return;
+        }
+
+        this.updateItems(items);
+      }
+      this.toggleEditMode(false);
+    },
+
+    /**
+     * Delete item and update items list
+     */
+    deleteItem(item?: string): number {
+      let index = -1;
+
+      if (item) {
+        index = findStringIndex(this.items as string[], item, false);
+        const items = this.items.filter(f => f !== item) as string[];
+
+        this.updateItems(items);
+      }
+
+      return index;
+    },
+
+    /**
+     * Update items list and emit to parent component
+     */
+    updateItems(items: string[]) {
+      this.$emit('change', items);
+    },
+  },
+
+});
+</script>
+
+<template>
+  <div
+    class="string-list"
+    :class="{ readonly }"
+  >
+    <div
+      ref="box"
+      data-testid="div-string-list-box"
+      class="string-list-box"
+      tabindex="0"
+      @dblclick="onClickEmptyBody()"
+    >
+      <div
+        v-for="item in items"
+        :key="item"
+        :ref="item"
+        :class="{
+          selected: selected === item,
+          readonly
+        }"
+        :data-testid="`div-item-${item}`"
+        class="item static"
+        tabindex="0"
+        @mousedown="onSelect(item)"
+        @dblclick.stop="toggleEditMode(true, item)"
+        @keydown.down.prevent="onSelectNext('down')"
+        @keydown.up.prevent="onSelectNext('up')"
+        @blur="onSelectLeave(item)"
+      >
+        <span
+          v-if="!isEditItem || isEditItem !== item"
+          class="label static"
+        >
+          {{ item }}
+        </span>
+        <input
+          v-if="isEditItem && isEditItem === item"
+          ref="item-edit"
+          class="input-field edit-input static"
+          :value="value != null ? value : item"
+          @input.prevent="onChange($event.target.value)"
+          @blur.prevent="toggleEditMode(false)"
+          @keydown.enter.prevent="updateItem(item)"
+        />
+      </div>
+      <div
+        v-if="isCreateItem"
+        class="create-field static"
+      >
+        <input
```

Comment:
Finally I would recommend the use of existing `LabeledInput` as input components, to keep consistency along the library for both visual and functionalities.

Reply:
I'll try to introduce the `LabeledInput`